l10n_th_vat_report/reports/vat_report.py

```diff
@@ -46,21 +46,27 @@ class VatReport(models.TransientModel):
     def _compute_results(self):
         self.ensure_one()
         self._cr.execute("""
-            SELECT aml.id as id, am.company_id, am.name, aml.account_id,
-                aml.tax_invoice, aml.partner_id, aml.date, aml.tax_date,
-                CASE WHEN ai.type in ('out_refund', 'in_refund')
-                then -aml.tax_base_amount
-                else aml.tax_base_amount end as tax_base_amount,
-                CASE WHEN aa.internal_group = 'asset'
-                then aml.balance else -aml.balance end as tax_amount
-            FROM account_move_line aml
-            JOIN account_move am on aml.move_id = am.id
-            JOIN account_account aa on aa.id = aml.account_id
-            LEFT JOIN account_invoice ai on ai.id = aml.invoice_id
-            WHERE aml.tax_line_id is not null and aml.date >= %s and
-                aml.date <= %s and aml.company_id = %s and
-                aml.account_id = %s and am.state = 'posted'
-            ORDER BY aml.tax_date, am.name
+            SELECT id, company_id, name, account_id, tax_invoice, partner_id,
+                date, tax_date, tax_amount,
+                CASE when tax_amount < 0 then -tax_base_amount
+                else tax_base_amount end as tax_base_amount
+            FROM (
+                SELECT aml.id as id, am.company_id, am.name, aml.account_id,
+                    aml.tax_invoice, aml.partner_id, aml.date, aml.tax_date,
+                    abs(aml.tax_base_amount) as tax_base_amount,
+                    CASE WHEN tx.type_tax_use = 'sale'
+                    then -aml.balance
```

Comment:
```suggestion
                    THEN -aml.balance
                    WHEN tx.type_tax_use = 'adjustment' THEN abs(aml.balance)
```

Supported Tax adjustment

Reply:
Done.

Reply:
@Saran440 can I merge?